src/render.ts

```diff
@@ -45,6 +45,83 @@ function handleCache(template: string | TemplateFunction, options: EtaConfig): T
+ *
+ * If `config.async` is `false`, Eta will return the rendered template.
+ *
+ * If `config.async` is `true` and there's a callback function, Eta will call the callback with `(err, renderedTemplate)`.
+ * If `config.async` is `true` and there's not a callback function, Eta will return a Promise that resolves to the rendered template.
+ *
+ * If `config.cache` is `true` and `config` has a `name` or `filename` property, Eta will cache the template on the first render and use the cached template for all subsequent renders.
+ *
+ * @param template Template string or template function
+ * @param data Data to render the template with
+ * @param config Optional config options
+ */
+export default function render(
+    template: string | TemplateFunction,
+    data: object,
+    config: PartialAsyncConfig
+): Promise<string>
+
+/**
+ * Render a template
+ *
+ * If `template` is a string, Eta will compile it to a function and then call it with the provided data.
+ * If `template` is a template function, Eta will call it with the provided data.
+ *
+ * If `config.async` is `false`, Eta will return the rendered template.
+ *
+ * If `config.async` is `true` and there's a callback function, Eta will call the callback with `(err, renderedTemplate)`.
+ * If `config.async` is `true` and there's not a callback function, Eta will return a Promise that resolves to the rendered template.
+ *
+ * If `config.cache` is `true` and `config` has a `name` or `filename` property, Eta will cache the template on the first render and use the cached template for all subsequent renders.
+ *
+ * @param template Template string or template function
+ * @param data Data to render the template with
+ * @param config Optional config options
+ */
+export default function render(
+    template: string | TemplateFunction,
+    data: object,
+    config?: PartialConfig
+): string
+
+/**
+ * Render a template
+ *
+ * If `template` is a string, Eta will compile it to a function and then call it with the provided data.
+ * If `template` is a template function, Eta will call it with the provided data.
+ *
+ * If `config.async` is `false`, Eta will return the rendered template.
+ *
+ * If `config.async` is `true` and there's a callback function, Eta will call the callback with `(err, renderedTemplate)`.
+ * If `config.async` is `true` and there's not a callback function, Eta will return a Promise that resolves to the rendered template.
+ *
+ * If `config.cache` is `true` and `config` has a `name` or `filename` property, Eta will cache the template on the first render and use the cached template for all subsequent renders.
+ *
+ * @param template Template string or template function
+ * @param data Data to render the template with
+ * @param config Optional config options
+ * @param cb Callback function
+ */
+export default function render(
+    template: string | TemplateFunction,
+    data: object,
+    config?: PartialConfig,
+    cb?: CallbackFn
+): string | Promise<string> | void
```

Comment:
I think we can delete lines 119-124.

Reply:
Removing the implementation signature from the function overloads would introduce a breaking change, for example the following code will not compile anymore (similar to above):

```ts
const optionalCallbackFn: CallbackFn | undefined = ...;
Eta.render(templateFn, data, { ...config, async: true }, optionalCallbackFn);
```

That is why I initially kept it, but if you want I can remove it of course.